winlogbeat/eventlog/eventlog.go

```diff
@@ -80,53 +82,66 @@ type Record struct {
-		"computer_name": e.Computer,
-		"record_number": strconv.FormatUint(e.RecordID, 10),
+	// Windows Log Specific data
+	win := common.MapStr{
+		"channel":       e.Channel,
 		"event_id":      e.EventIdentifier.ID,
+		"provider_name": e.Provider.Name,
+		"record_id":     e.RecordID,
+		"task":          e.Task,
+		"api":           e.API,
 	}
-
-	addOptional(m, "xml", e.XML)
-	addOptional(m, "provider_guid", e.Provider.GUID)
-	addOptional(m, "version", e.Version)
-	addOptional(m, "level", e.Level)
-	addOptional(m, "task", e.Task)
-	addOptional(m, "opcode", e.Opcode)
-	addOptional(m, "keywords", e.Keywords)
-	addOptional(m, "message", sys.RemoveWindowsLineEndings(e.Message))
-	addOptional(m, "message_error", e.RenderErr)
-
+	addOptional(win, "computer_name", e.Computer)
+	addOptional(win, "kernel_time", e.Execution.KernelTime)
+	addOptional(win, "keywords", e.Keywords)
+	addOptional(win, "opcode", e.Opcode)
+	addOptional(win, "processor_id", e.Execution.ProcessorID)
+	addOptional(win, "processor_time", e.Execution.ProcessorTime)
+	addOptional(win, "provider_guid", e.Provider.GUID)
+	addOptional(win, "session_id", e.Execution.SessionID)
+	addOptional(win, "task", e.Task)
+	addOptional(win, "user_time", e.Execution.UserTime)
+	addOptional(win, "version", e.Version)
+	addOptional(win, "message_error", e.RenderErr)
 	// Correlation
-	addOptional(m, "activity_id", e.Correlation.ActivityID)
-	addOptional(m, "related_activity_id", e.Correlation.RelatedActivityID)
-
+	addOptional(win, "activity_id", e.Correlation.ActivityID)
+	addOptional(win, "related_activity_id", e.Correlation.RelatedActivityID)
 	// Execution
-	addOptional(m, "process_id", e.Execution.ProcessID)
-	addOptional(m, "thread_id", e.Execution.ThreadID)
-	addOptional(m, "processor_id", e.Execution.ProcessorID)
-	addOptional(m, "session_id", e.Execution.SessionID)
-	addOptional(m, "kernel_time", e.Execution.KernelTime)
-	addOptional(m, "user_time", e.Execution.UserTime)
-	addOptional(m, "processor_time", e.Execution.ProcessorTime)
+	addOptional(win, "process.pid", e.Execution.ProcessID)
+	addOptional(win, "process.thread.id", e.Execution.ThreadID)
 
 	if e.User.Identifier != "" {
 		user := common.MapStr{
 			"identifier": e.User.Identifier,
 		}
-		m["user"] = user
-
+		win["user"] = user
 		addOptional(user, "name", e.User.Name)
 		addOptional(user, "domain", e.User.Domain)
 		addOptional(user, "type", e.User.Type.String())
 	}
 
-	addPairs(m, "event_data", e.EventData.Pairs)
-	userData := addPairs(m, "user_data", e.UserData.Pairs)
+	addPairs(win, "event_data", e.EventData.Pairs)
+	userData := addPairs(win, "user_data", e.UserData.Pairs)
 	addOptional(userData, "xml_name", e.UserData.Name.Local)
 
+	m := common.MapStr{
+		"winlog": win,
+	}
+
+	// ECS data
+	m.Put("event.kind", "event")
+	m.Put("event.dataset", fmt.Sprintf("%v.%v", e.API, strings.ToLower(e.Channel)))
+	m.Put("event.code", e.EventIdentifier.ID)
+	addOptional(m, "event.action", e.Task)
+
+	m.Put("host.hostname", e.Computer)
```

Comment:
Winlogbeat can't accept remotely, but the Windows Event Log subsystem can do federation. So it may in fact be processing messages that come from remote hosts. I should double-check if Computer will always be the host at the origin of the event.

(host 1: Winlog) => (host 2: Winlog => Winlogbeat)

Reply:
Won't this `host.hostname` value get overwritten by the libbeat publisher pipeline?

Like this stuff https://github.com/elastic/beats/blob/2ec5e1aa8cc82ced325088f8ad225afbe3d4c8bf/libbeat/publisher/pipeline/processor.go#L130-L134.